modules/migrate.py

```diff
@@ -0,0 +1,485 @@
+    configuration file.
+    """
+    results = []
+    if args.config:
+        with open(args.config, 'rb') as f:
+            reader = csv.reader(f)
+            try:
+                for row in reader:
+                    # Strip any whitespace in row contents.
+                    row = map(str.strip, row)
+                    if len(row) < 4 or not (row[0] and row[1] and row[2] and
+                            row[3]):
+                        log.warning('CSV has invalid number of arguments at '
+                            'line %s, skipping line.\n', reader.line_num)
+                    else:
+                        results.append((row[0], row[1], row[2], row[3]))
+            except csv.Error as e:
+                log.error('Unable to parse CSV at line %s, skipping rest of'
+                    ' file.\n', reader.line_num)
+
+    return results
+
+
+def _getAllHTMLFiles(rootDir):
+    """Returns a list of all HTML files under the root directory.
+
+    Args:
+        rootDir - String absolute path of the directory tree to search.
+    """
+    htmlFiles = []
+    for root, dirs, files in os.walk(rootDir):
+        for filename in [f for f in files if f.endswith('.html')]:
+            htmlFiles.append(os.path.join(root, filename))
+    return htmlFiles
+
+def _updateHTMLFile(filePath, widgetAbsolutePath, modularWidgetAbsolutePath):
+    """Updates file and linked widget JSON config files.
+
+    Updates the specified HTML file changing all references to the
+    non-modular widget to instead reference the modular widget, and all paths
+    relative to the non-modular widget to be relative to the modular widget.
+    Also fixes any JSON configuration files references from a widget's <object>
+    tag.
+    """
+    log.debug('Reading HTML file: %s', filePath)
+    with open(filePath, 'rb') as htmlFile:
+        lines = ''.join(htmlFile.readlines())
+        soup = BeautifulSoup(lines)
+
+        # Non-modular Widget view page path relative to the content. This path
+        # will be the data attribute of any widget object tags.
+        widgetViewRelativePath = os.path.join(
+            os.path.relpath(widgetAbsolutePath, os.path.dirname(filePath)),
+            'index.html')
+
+        # Absolute path to the modular version of the widget view page.
+        absModuleViewPath = os.path.join(
+            modularWidgetAbsolutePath, 'index.html')
+        # Widget view page path relative to the content.
+        relativeModuleViewPath = os.path.relpath(
+            absModuleViewPath, os.path.dirname(filePath))
+
+        # Convert all object data (iframe src) links to new modular location.
+        # This migrates all the widgets to point to their modular versions, but
+        # leaves all parameters and JSON config files incorrectly relative to
+        # the wrong location.
+        lines = re.sub(r'(<object\s.*?data=")' + widgetViewRelativePath +'"',
+                       r'\g<1>' + relativeModuleViewPath + '"',
```

Comment:
In the regex below you extracted the first argument to a variable called `toReplace`. I found that really helpful. Could you also extract the second argument to another variable called maybe `replacement`.